perceval/components/core_catalog/toffoli.py

```diff
@@ -26,42 +26,34 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-from deprecated import deprecated
-from perceval.components import Circuit, Unitary, Port
+from perceval.components import Circuit, Port, BS
 from perceval.components.component_catalog import CatalogItem, AsType
-from perceval.utils import Encoding, Matrix, PostSelect
+from perceval.components.core_catalog.postprocessed_ccz import PostProcessedCCZItem
+from perceval.utils import Encoding, PostSelect
 
 
 class ToffoliItem(CatalogItem):
-    description = r"""Toffoli gate (CCNOT) with 6 heralded modes"""
-    str_repr = r"""                       ╭─────╮
-ctrl0 (dual rail) ─────┤     ├───── ctrl0 (dual rail)
-                  ─────┤     ├─────
-                       │     │
-ctrl1 (dual rail) ─────┤     ├───── ctrl1 (dual rail)
-                  ─────┤     ├─────
-                       │     │
-data (dual rail)  ─────┤     ├───── data (dual rail)
-                  ─────┤     ├─────
-                       ╰─────╯"""
+    description = r"""Toffoli gate CCNOT gate with 6 heralded modes and a post-selection function (built using Heralded CCZ and H)."""
+    str_repr = r"""                               ╭──────────╮
+ctrl0 (dual rail) ─────────────┤          ├───────────── ctrl0 (dual rail)
+                  ─────────────┤          ├─────────────
+ctrl1 (dual rail) ─────────────┤          ├───────────── ctrl1 (dual rail)
+                  ─────────────┤ Heralded ├─────────────
```

Comment:
Toffoli is not based on a "Heralded CZ" but on a "Post-processed CCZ"!

Reply:
Actually It's a heralded & post-processed ccz but i will change to just "ccz"